server/controllers/User/oauth.js

```diff
@@ -0,0 +1,45 @@
+const { OAuth2Client } = require('google-auth-library');
+const { SignToken } = require('../../utils/jwt/jwt');
+
+const { GOOGLE_ID } = process.env;
+const User = require('../../models/user');
+
+const client = new OAuth2Client(GOOGLE_ID);
+
+const oauth = async (req, res) => {
+  const { token: googleToken } = req.body;
+  try {
+    const ticket = await client.verifyIdToken({
+      idToken: googleToken,
+      audience: GOOGLE_ID,
+    });
+
+    const { email, name } = ticket.getPayload();
+
+    const existingUser = User.findOne({
+      where: {
+        email,
+      },
+    });
+    if (!existingUser) {
+      // eslint-disable-next-line no-unused-vars
```

Comment:
if you use error first approach you dont need 
      // eslint-disable-next-line no-unused-vars
      anymore